src/types/basic.rs

```diff
@@ -49,41 +54,93 @@ impl TypeClass for ClassicLeaf {
 impl TypeClass for AnyLeaf {
     const TAG: TypeTag = TypeTag::Simple;
 }
-pub struct TaggedWrapper<T>(TypeTag, T);
 
-impl<T> TaggedWrapper<T> {
-    pub fn tag(&self) -> TypeTag {
-        self.0
+#[derive(Clone, PartialEq, Debug, Eq)]
+pub struct Tagged<I, T>(I, PhantomData<T>);
+
+pub trait GetTag {
```

Comment:
I'd call this Tagged or DynamicTag

Reply:
RuntimeTag? Maybe just Tagged, the statically-known one is called type_tag_bound() after all

Reply:
Rust being Rust, if this is GetTag then the method should be `get_tag`. I am *not* saying I like this....

but that'd work for RuntimeTag or DynamicTag, tho, if you can think of a better adjective than Runtime/Dynamic